### DISCUSSION TITLE
Checking 3 values in rate check

### DISCUSSION THREAD

**Discussion by amandel** [jomjol/AI-on-the-edge-device#2790] [Ideas]
Hi,

I've setup a new device, from time to time the image shows a split or wrap-around error. I'm still looking for the reason. Due to good alignment this leads to false readings mainly in the analog area. Sometimes the read values are well inside the rate check, but still wrong. This leads to a "Neg. Rate" with the next rounds until the actual value increased beyond the false reading.

The idea is consider 3 values for the rate check before accepting the value: (1) Previous Value, (2) Value last measurement, (3) Value this measurement. Then accept the value of the last measurement (2) only if the rate check works for (3) too. This would catch the case, where (2) is a false reading:

1: 0.1234
2: 0.1255
3: 0.1234

The drawback surely is that the values can be reported only with the latency of 1 round interval. 

Sorry if this had been suggested already. Since I've only 1 device I can not tell if this is a common issue or it is a coincidence that hits just myself.

**Comment by caco3** [comment]
There is no such functionality yet, but contributions are welcome.

I also see such thing sporadically, mainly on the last digit which gets interpreted as flipped to early. 
There are some parameters to tweak this, but I still have it every 1m3.
